polyfills/systemxhr/systemxhr.js

```diff
@@ -0,0 +1,430 @@
+      navConnPromise.methodCall({
+                                 methodName: 'overrideMimeType',
+                                 numParams: 1,
+                                 returnValue: VoidRequest,
+                                 promise: _systemxhr,
+                                 field: 'xhrId'
+                               }, mymetype);
+    };
+
+    // Null if no response has been received yet
+    this._responseHeaders = null;
+    this.getResponseHeader = function(header) {
+      return this._responseHeaders[header] ?
+        this._responseHeaders[header] : null;
+    };
+
+    this.getAllResponseHeaders = function() {
+      var headers = '';
+
+      if (!this._responseHeaders) {
+        return null;
+      }
+
+      for (var header in this._responseHeaders) {
+        headers += header + ': ' + this._responseHeaders[header] + ' \n';
+      }
+
+      return headers;
+    };
+
+    this.send = function(data) {
+      navConnPromise.methodCall({
+                                 methodName: 'send',
+                                 numParams: 1,
+                                 returnValue: VoidRequest,
+                                 promise: _systemxhr,
+                                 field: 'xhrId'
+                               }, data);
+    };
+
+    this.setRequestHeader = function(header, value) {
+      navConnPromise.methodCall({
+                                 methodName: 'setRequestHeader',
+                                 numParams: 2,
+                                 returnValue: VoidRequest,
+                                 promise: _systemxhr,
+                                 field: 'xhrId'
+                               }, header, value);
+    };
+
+    var readyStates = [
+      {property: 'UNSENT', value: 0},
+      {property: 'OPENED', value: 1},
+      {property: 'HEADERS_RECEIVED', value: 2},
+      {property: 'LOADING', value: 3},
+      {property: 'DONE', value: 4},
+    ];
+
+    readyStates.forEach(state => {
+      Object.defineProperty(this, state.property, {
+        enumerable: true,
+        value: state.value
+      });
+    });
+
+    this.upload = new FakeXMLHttpRequestUpload();
+
+    var readOnlyProperties = [
+      'response', 'responseText', 'responseType', 'responseXML',
+      'status', 'statusText', 'readyState',' timeout', 'responseURL'
+    ];
+
+    readOnlyProperties.forEach(property => {
+      Object.defineProperty(this, property, {
+        enumerable: true,
+        get: function() {
+          return this['_' + property];
+        }
+      });
+    });
+
+    Object.defineProperty(this, 'onreadystatechange', {
+      get: function() {
+        return this._onreadystatechangecb;
+      },
+
+      set: function(cb) {
+        this._onreadystatechangecb = cb;
+      }
+    });
+
+    // Need to listen to readyState events even if the callback is not defined
+    this._onreadystatechange = navConnPromise.methodCall.bind(navConnPromise,
```

Comment:
Adding a returnValue field to the list of methods, this can be done also on the forEach of the first comment